lib/Model/Importer/Binary/ELFImporter.cpp

```diff
@@ -326,9 +328,76 @@ Error ELFImporter<T, HasAddend>::import(bool FetchDebugInfo) {
   DwarfImporter Importer(Model);
   Importer.import(TheBinary.getFileName(), FetchDebugInfo);
 
+  // Now we try to find missing types in the dependencies.
+  if (FetchDebugInfo > 1)
+    findMissingTypes(TheELF, FetchDebugInfo);
+
   return Error::success();
 }
 
+template<typename T, bool HasAddend>
+void ELFImporter<T, HasAddend>::findMissingTypes(object::ELFFile<T> &TheELF,
+                                                 unsigned FetchDebugInfo) {
+  ModelMap ModelsOfLibraries;
+
+  LDDTree Dependencies;
+  lddtree(Dependencies, TheBinary.getFileName().str(), FetchDebugInfo);
+  for (auto &Library : Dependencies) {
+    revng_log(ELFImporterLog,
+              "Importing Models for dependencies of " << Library.first
+                                                      << ":\n");
+    for (auto &DependencyLibrary : Library.second) {
+      if (ModelsOfLibraries.count(DependencyLibrary))
+        continue;
+      revng_log(ELFImporterLog,
+                " Importing Model for: " << DependencyLibrary << "\n");
+      auto BinaryOrErr = llvm::object::createBinary(DependencyLibrary);
+      if (not BinaryOrErr) {
+        llvm::consumeError(BinaryOrErr.takeError());
+        revng_log(ELFImporterLog,
+                  "Can't create object for " << DependencyLibrary);
+        return;
```

Comment:
Why bail out entirely? Can't we just skip this one?

Reply:
it makes sense